ts/indicators/vwap.src.ts

```diff
@@ -0,0 +1,253 @@
+             * OHLC data, volumeSeriesID='volume' means the indicator will be
+             * calculated using OHLC and volume values.
+             */
+            volumeSeriesID: 'volume'
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        /**
+         * Returns the final values of the indicator ready to be presented on a
+         * chart
+         * @private
+         * @param {Highcharts.VWAPIndicator} this indicator
+         * @param {Highcharts.Series} series - series for indicator
+         * @param {object} params - params
+         * @return {object} - computed VWAP
+         **/
+        getValues: function (
+            this: Highcharts.VWAPIndicator,
+            series: Highcharts.VWAPLinkedParentSeries,
+            params: Highcharts.VWAPIndicatorParamsOptions,
+        ): Highcharts.IndicatorValuesObject {
+            var indicator = this,
+                chart: Highcharts.Chart = series.chart,
+                xValues: Array<number> = series.xData,
+                yValues: (
+                    Array<number>|Array<[number, number, number, number]>
+                ) = series.yData,
+                period: number = (params.period as any),
+                isOHLC = true,
+                volumeSeries: Highcharts.Series;
+
+            // Checks if volume series exists
+            if (!(volumeSeries = (
+                chart.get(params.volumeSeriesID as any)) as any
+            )) {
+                return (H.error(
+                    'Series ' +
+                    params.volumeSeriesID +
+                    ' not found! Check `volumeSeriesID`.',
+                    true,
+                    chart
+                ) as any);
+            }
+
+            // Checks if series data fits the OHLC format
+            if (!(isArray(yValues[0]))) {
+                isOHLC = false;
+            }
+
+            return indicator.calculateVWAPValues(
+                isOHLC,
+                xValues,
+                yValues,
+                volumeSeries,
+                period
+            );
+        },
+        /**
+         * Main algorithm used to calculate Volume Weighted Average Price (VWAP)
+         * values
+         * @private
+         * @param {boolean} isOHLC - says if data has OHLC format
+         * @param {Array<number>} xValues - array of timestamps
+         * @param {Array<number|Array<number,number,number,number>>} yValues -
+         * array of yValues, can be an array of a four arrays (OHLC) or array of
+         * values (line)
+         * @param {Array<*>} volumeSeries - volume series
+         * @param {number} period - number of points to be calculated
+         * @return {object} - Object contains computed VWAP
+         **/
+        calculateVWAPValues: function (
+            isOHLC: boolean,
+            xValues: Array<number>,
+            yValues: (Array<number>|Array<[number, number, number, number]>),
```

Comment:
Added type as described in jsdoc rather than `Array<Array<number>>. This indicator works with both structures.`